menas/src/main/scala/za/co/absa/enceladus/menas/controllers/SchemaController.scala

```diff
@@ -62,16 +78,18 @@ class SchemaController @Autowired()(
                        @RequestParam version: Int,
                        @RequestParam name: String,
                        @RequestParam format: Optional[String]): CompletableFuture[Option[Schema]] = {
-    val origFormat: Option[String] = format
+
     val fileContent = new String(file.getBytes)
+    val origFormat: Option[String] = format
 
     val (sparkStruct, schemaType) = origFormat match {
-      case Some(SchemaTypeStruct) | Some("") | None => (schemaParsers.parseStructType(fileContent), SchemaTypeStruct)
-      case Some(SchemaTypeCopybook) => (schemaParsers.parseCopybook(fileContent), SchemaTypeCopybook)
-      case Some(SchemaTypeAvro) => (schemaParsers.parseAvro(fileContent), SchemaTypeAvro)
-      case Some(schemaType) =>
-        throw SchemaFormatException(schemaType, s"'$schemaType' is not a recognized schema format. " +
-          s"Menas currently supports: '$SchemaTypeStruct' and '$SchemaTypeCopybook'.")
+      case Some("") | None => (schemaParsers.parseStructType(fileContent), SchemaType.Struct) // legacy compatibility cases
+      case Some(explicitSchemaName) =>
+        SchemaType.fromSchemaName(explicitSchemaName) match {
```

Comment:
The "double match" structure looks a bit weird, but the most obvious way of writing it with
```
case Some(Struct) => ...
case Some(Copybook) => ...
case Some(Avro) => ...
```
would fail on being exhaustive, which would need to introduce another default case/error handling -- that looked even worse.